src/Aspire.Dashboard/Otlp/Http/OtlpHttpEndpointsBuilder.cs

```diff
@@ -0,0 +1,205 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Buffers;
+using System.IO.Pipelines;
+using System.Net.Http.Headers;
+using System.Reflection;
+using Aspire.Dashboard.Authentication;
+using Google.Protobuf;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.Extensions.Primitives;
+using OpenTelemetry.Proto.Collector.Logs.V1;
+using OpenTelemetry.Proto.Collector.Metrics.V1;
+using OpenTelemetry.Proto.Collector.Trace.V1;
+
+namespace Aspire.Dashboard.Otlp.Http;
+
+public static class OtlpHttpEndpointsBuilder
+{
+    public const string ProtobufContentType = "application/x-protobuf";
+    public const string JsonContentType = "application/json";
+
+    public static void MapHttpOtlpApi(this IEndpointRouteBuilder endpoints)
```

Comment:
It's typically recommended that `MapX` implementations like this one return the `IEndpoitRouteBuilder` instance passed in in order to support chaining invocations onto the call (e.g. adding metadata via `WithMetadta`). Is there a particular reason to prohibit chaining on this API?

Reply:
This is only called in on place. It’s not a library, just part of an app. It’s less important in that context